src/core/mesh/qgstriangularmesh.h

```diff
@@ -124,6 +124,17 @@ class CORE_EXPORT QgsTriangularMesh // TODO rename to QgsRendererMesh in QGIS 4
      */
     int faceIndexForPoint( const QgsPointXY &point ) const ;
 
+    /**
+     * Finds index of triangle at given point
+     * It uses spatial indexing and don't use geos to be faster
+     *
+     * \param point point in map coordinate system
+     * \returns triangle index that contains the given point, -1 if no such triangle exists
+     *
+     * \since QGIS 3.12
```

Comment:
do we want to deprecate faceIndexForPoint? also isn't since 3.22 more appropriate (not 3.12?)

Reply:
this method is here since 3.12. Yes, I think we can deprecate the previous version. Or replace it by the newer implementation and deprecate `faceIndexForPoint_v2()` ?